include/amici_hdf5.h

```diff
@@ -2,28 +2,49 @@
 #define AMICI_HDF5_H
 
 #include <string>
+#include <memory>
+#include <vector>
 
-#include <hdf5.h>
-#include <hdf5_hl.h>
+#include <H5Cpp.h>
 
 #undef AMI_HDF5_H_DEBUG
 
+#define AMICI_H5_SAVE_ERROR_HANDLER                                            \
+    herr_t (*old_func)(void *);                                                \
+    void *old_client_data;                                                     \
+    H5Eget_auto1(&old_func, &old_client_data);                                 \
+    H5Eset_auto1(NULL, NULL)
+
+#define AMICI_H5_RESTORE_ERROR_HANDLER H5Eset_auto1(old_func, old_client_data)
+
 namespace amici {
 
 class ReturnData;
 class ExpData;
 class Model;
 class Solver;
 
-/* Functions for reading and writing AMICI data to/from HDF5 files. */
+namespace hdf5 {
+
+
+/** @file Functions for reading and writing AMICI data to/from HDF5 files. */
+
 // TODO: proper type checking, exception instead of return code, c++ api; check if all fields are read and saved
```

Comment:
this is no longer TBD right?

Reply:
at least partially

Reply:
Partially. Let's leave it until we changed the rest and adapted to new storage format.